### PR TITLE
Fix ros2 launch

### PR DESCRIPTION
**Pull request by ahcorde**
As @dirk-thomas indicates in this comment 

> argcomplete it doesn't support Windows. 

Moved the `argcomplete` outside the try catch.

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Please run code you change to confirm it works as expected. `FilesCompleter` is being used in the module and will raise if not imported.

**Comment by ahcorde** [comment]
I tried on Windows and Ubuntu Focal. 

is it enough to run CI up-to ros2lanch and run only these tests?

**Comment by dirk-thomas** [comment]
> is it enough to run CI up-to ros2lanch and run only these tests?

I don;t think any unit test uses `ros2 launch`. So the CI build will only be for linters.

**Comment by ahcorde** [comment]
I ckecked other PR related with argcomplete https://github.com/ros2/launch_ros/pull/126, it's only testing ros2launch. I also checked `system_test` repository, but it's only testing parameters and remappings. 

CI (testing just ros2launch)

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10659/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6086/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8674/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10567/)

**Comment by jacobperron** [comment]
I think this fixes an issue on all platforms (not just Windows). I've run into it on Linux. The code before this patch assumes that argcomplete is installed, but it is [optional according to our docs](https://index.ros.org/doc/ros2/Installation/Foxy/Linux-Install-Debians/#install-argcomplete-optional). Trying `ros2 launch` (or even `ros2 --help`) on a fresh installation results in the error:

```
Failed to load entry point 'launch': No module named 'argcomplete'
```

File: ros2launch/ros2launch/command/launch.py [hunk 1]
```diff
@@ -16,7 +16,11 @@
 
 from ament_index_python.packages import get_package_prefix
 from ament_index_python.packages import PackageNotFoundError
-from argcomplete.completers import FilesCompleter
+try:
+    from argcomplete.completers import FilesCompleter
+except ImportError:
+    # argcomplete is not supported on Windows
```
Comment (jacobperron) [inline ros2launch/ros2launch/command/launch.py:22]: nitpick: A better comment might indicate that argcomplete is optional

```suggestion
    # argcomplete is optional
```

**Comment by jacobperron** [comment]
I can confirm this PR resolves my issue on Linux.

**Comment by ros-discourse** [comment]
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/ros-2-foxy-fitzroy-call-for-testing-and-package-releases/13998/1

**Comment by ahcorde** [comment]
thank you @jacobperron

**Pull request merged**